anesthetic/tension.py

```diff
@@ -0,0 +1,90 @@
+"""Tension statistics between two datasets."""
+from anesthetic.samples import Samples
+from scipy.stats import chi2
+
+
+def tension_stats(A, B, AB, nsamples=None, beta=None):  # noqa: D301
+    """Compute tension statistics between two samples.
+
+    Using nested sampling we can compute:
+
+    - ``logZ``: Bayesian evidence
+
+      .. math::
+          \\log Z = \\log \\int L \\pi d\\theta
+
+    - ``D_KL``: Kullback--Leibler divergence
+
+      .. math::
+          D_\\mathrm{KL} = \\int P \\log(P / \\pi) d\\theta
+
+    - ``logL_P``: posterior averaged log-likelihood
+
+      .. math::
+          \\langle\\log L\\rangle_P = \\int P \\log L d\\theta
+
+    - ``d_G``: Gaussian model dimensionality
+      (or posterior variance of the log-likelihood)
+
+      .. math::
+          d_\\mathrm{G}/2 = \\mathrm{var}(\\log L)_P
+
+    - ``p``: p-value for the tension between two samples
+
+      .. math::
+          p = \\int_{d_\\mathrm{G} - 2 \\log S}^{\\infty} \\chi^2 (x)dx
+
+    Parameters
+    ----------
+    A : :class:`anesthetic.samples.Samples` or \
+        :class:`anesthetic.samples.NestedSamples`
```

Comment:
Currently this only works for `NestedSamples` and not for `Samples`. 
If we want this to work for general `Samples`, then we should add the necessary modifications to `samples.py`. 
Otherwise the type hinting should only list `NestedSamples`.

(Side note: outside `samples.py` you have to use ``:class:`anesthetic.samples.Samples` `` rather than  ``:class:`Samples` `` for the automatic linking to work.)

Reply:
Let's leave this to a later date.

I've updated the docstrings to include \\mathcals, and in addition I have renamed tension_stats to just stats, since `anesthetic.tension.tension_stats` is a bit repetitive.